// ngx-notification/esm5/lib/ngx-notification.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
var NgxNotificationService = /** @class */ (function () {
    function NgxNotificationService() {
        this.subject = new Subject();
    }
    /**
     * @param {?} message
     * @param {?} theme
     * @param {?} position
     * @return {?}
     */
    NgxNotificationService.prototype.sendMessage = /**
     * @param {?} message
     * @param {?} theme
     * @param {?} position
     * @return {?}
     */
    function (message, theme, position) {
        this.subject.next({
            text: message,
            theme: theme,
            position: position
        });
    };
    /**
     * @return {?}
     */
    NgxNotificationService.prototype.clearMessage = /**
     * @return {?}
     */
    function () {
        this.subject.next();
    };
    /**
     * @return {?}
     */
    NgxNotificationService.prototype.getMessage = /**
     * @return {?}
     */
    function () {
        return this.subject.asObservable();
    };
    NgxNotificationService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    NgxNotificationService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgxNotificationService.ngInjectableDef = i0.defineInjectable({ factory: function NgxNotificationService_Factory() { return new NgxNotificationService(); }, token: NgxNotificationService, providedIn: "root" });
    return NgxNotificationService;
}());
export { NgxNotificationService };
function NgxNotificationService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxNotificationService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxNotificationService.ctorParameters;
    /** @type {?} */
    NgxNotificationService.prototype.subject;
}
