// ngx-notification/esm5/lib/ngx-notification.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component } from '@angular/core';
import { NgxNotificationService } from './ngx-notification.service';
var NgxNotificationComponent = /** @class */ (function () {
    function NgxNotificationComponent(ngxNotificationService) {
        var _this = this;
        this.ngxNotificationService = ngxNotificationService;
        this.message = {};
        this.subscription = this.ngxNotificationService.getMessage()
            .subscribe(function (message) {
            _this.show = true;
            _this.message = message;
            console.log(_this.message);
            _this.autoHideMessage();
        });
    }
    /**
     * @return {?}
     */
    NgxNotificationComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    NgxNotificationComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.subscription.unsubscribe();
    };
    /**
     * @return {?}
     */
    NgxNotificationComponent.prototype.autoHideMessage = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout(function () {
            _this.show = false;
        }, 5000);
    };
    NgxNotificationComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-ngx-notification',
                    template: "<div *ngIf='show' class='ngx-notification {{ message.position }} {{ message.theme }} show'>{{message?.text }}</div>",
                    styles: ["\n  .ngx-notification.dark {\n    background-color: #2d2d30;\n    border-color: #272729;\n    color: #c9c9c9;\n  }\n  .ngx-notification.light {\n    background-color: #ffffff;\n    border-color: #cccccc;\n    color: #2d2d30;\n  }\n  .ngx-notification.success {\n    background-color: #dff0d8;\n      border-color: #d0e9c6;\n      color: #3c763d;\n  }\n  .ngx-notification.info {\n    background-color: #d9edf7;\n    border-color: #bcdff1;\n    color: #31708f;\n  }\n  .ngx-notification.warning {\n    background-color: #fcf8e3;\n    border-color: #faf2cc;\n    color: #8a6d3b;\n  }\n  .ngx-notification.danger {\n    background-color: #f2dede;\n    border-color: #ebcccc;\n    color: #a94442;\n  }\n  .ngx-notification.top-left {\n    position: fixed;\n    left: 10px;\n    top: 10px;\n  }\n  .ngx-notification.top-right {\n    position: fixed;\n    right: 10px;\n    top: 10px;\n  }\n  .ngx-notification.bottom-right {\n    position: fixed;\n    right: 10px;\n    bottom: 10px;\n  }\n  .ngx-notification.bottom-left {\n    position: fixed;\n    left: 10px;\n    bottom: 10px;\n  }\n  .ngx-notification.center {\n    position:absolute;\n      top: 50%;\n      left: 50%;\n      margin-top: -9em;\n      margin-left: -15em;\n  }\n  .ngx-notification {\n    padding: .75rem 1.25rem;\n    margin-bottom: 1rem;\n    border: 1px solid transparent;\n    border-radius: .25rem;\n    -webkit-animation: fadein 1.5s, fadeout 1.5s 3.5s;\n    animation: fadein 1.5s, fadeout 1.5s 3.5s;\n    z-index : 1;\n  }\n\n  /* Animations to fade the snackbar in and out */\n  @-webkit-keyframes fadein {\n    from {opacity: 0;}\n    to {opacity: 1;}\n  }\n\n  @keyframes fadein {\n    from {opacity: 0;}\n    to {opacity: 1;}\n  }\n\n  @-webkit-keyframes fadeout {\n    from {opacity: 1;}\n    to {opacity: 0;}\n  }\n\n  @keyframes fadeout {\n    from {opacity: 1;}\n    to {opacity: 0;}\n  }\n  "]
                },] },
    ];
    /** @nocollapse */
    NgxNotificationComponent.ctorParameters = function () { return [
        { type: NgxNotificationService, },
    ]; };
    return NgxNotificationComponent;
}());
export { NgxNotificationComponent };
function NgxNotificationComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxNotificationComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxNotificationComponent.ctorParameters;
    /** @type {?} */
    NgxNotificationComponent.prototype.show;
    /** @type {?} */
    NgxNotificationComponent.prototype.message;
    /** @type {?} */
    NgxNotificationComponent.prototype.subscription;
    /** @type {?} */
    NgxNotificationComponent.prototype.ngxNotificationService;
}
